### PR TITLE
[SSDK-495] Build with updated Search SDK v2

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description
Fixes [#SSDK-495](https://mapbox.atlassian.net/browse/SSDK-495)

#### Packages and Project

- Update MapboxCommon 23.6.0 -> 24.0.0
	- This will require MapboxMaps 11.0.0+
- Update MapboxCoreSearch 0.71.0 -> 2.0.0-alpha.4
- Update CI, fake, and Fastfile devices to supported OS and device combinations
- Updated to Xcode 14.1 minimum version
- Updated deployment target to iOS 12
- Remove bitcode support

#### SDK Changes

- Removed parameter-based Access Token. Be sure to provide your token in Info.plist.
- Renamed `CoreSuggestAction.isMultiRetrivable` to `multiRetrievable`.
- Renamed `CoreSearchResult.center` to `.centerLocation`.
- Renamed `CoreSearchOptions.isIgnoreUR` to `ignoreUR`.
- Renamed `TileRegionLoadOptions` initializer parameter `start` to `startLocation`.
- Replace some `CLLocation` fields with `Coordinate2D` wrapper containing a value of `CLLocationCoordinate2D`. This changes the call-site from `.coordinate` to `.value`.
- Added `SdkInformation.defaultInfo` default value for various Core initializer parameters.
- Added `SearchAddressRegion` containing `name`, `regionCode`, and `regionCodeFull` fields.
- Added `SearchAddressCountry` containing `name`, `countryCode`, and `regionCodeFull` fields.
- Added fields `searchAddressRegion` and `searchAddressCountry` to `Address` alongside existing `country` and `region`.
- Remove access token parameter from `SearchTileStore`.

#### Examples
- Update Examples/SearchExamples for Xcode 14.1, iOS 13, and SDK versions found on main branch (preceeding versions)
- Add tab icons to MapboxSearch.xcodeproj > Demo build target

### Checklist
- [x] Update `CHANGELOG`

### CODE REVIEW

File: Sources/MapboxSearch/PublicAPI/MapboxSearchVersion.swift [hunk 1]
```diff
@@ -1,2 +1,2 @@
 /// Mapbox Search SDK version variable
-public let mapboxSearchSDKVersion = "1.0.0-rc.7"
+public let mapboxSearchSDKVersion = "1.0.0-rc.8"
```
Comment (ahmedaly16) [inline Sources/MapboxSearch/PublicAPI/MapboxSearchVersion.swift:2]: My only concern is this line. Point #6 [here](https://mapbox.slack.com/archives/C05FS658HAR/p1702662408729689?thread_ts=1702662189.709229&cid=C05FS658HAR) from @DzmitryFomchyn was suggesting something else. As discussed offline I know this was auto generated, but probably we will need to update it manually. What do you think @DzmitryFomchyn?

File: Package.swift [hunk 2]
```diff
@@ -4,10 +4,10 @@
 import PackageDescription
 import Foundation
 
-let (coreSearchVersion, coreSearchVersionHash) = ("0.71.0", "252327e268abc28b9e5546a381d969c5cd1a3a8ccbd4b3ae2a31cef448a61fbc")
+let (coreSearchVersion, coreSearchVersionHash) = ("v2.0.0-alpha.4", "48a306d5d65b3eeef0a1498bd574cb79964868d13ccb299e57d075098a553866")
```
Comment (OdNairy) [inline Package.swift:7]:
```suggestion
let (coreSearchVersion, coreSearchVersionHash) = ("2.0.0-alpha.4", "48a306d5d65b3eeef0a1498bd574cb79964868d13ccb299e57d075098a553866")
```

File: Sources/MapboxSearch/PublicAPI/MapboxSearchVersion.swift (hunk 1 above)
Comment (aokj4ck) [reply Sources/MapboxSearch/PublicAPI/MapboxSearchVersion.swift]: This was resolved by SSDK-517 and merged in PR 142

**Review by globaltrouble — APPROVED** [review]

**Review by globaltrouble — APPROVED** [review]

File: Examples/SearchExamples.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -439,33 +437,38 @@
...
 		};
-		FE93032326F8D87A00644DF6 /* XCRemoteSwiftPackageReference "mapbox-maps-ios" */ = {
+		FE0720BC26FC86C70065A273 /* XCRemoteSwiftPackageReference "search-ios" */ = {
 			isa = XCRemoteSwiftPackageReference;
-			repositoryURL = "git@github.com:mapbox/mapbox-maps-ios.git";
+			repositoryURL = "https://github.com/mapbox/search-ios";
 			requirement = {
 				kind = upToNextMajorVersion;
-				minimumVersion = "10.0.0-rc";
+				minimumVersion = "1.0.0-rc.8";
```
Comment (azarovalex) [inline Examples/SearchExamples.xcodeproj/project.pbxproj:453]: Maybe this should be a local SPM package, or Examples app won't represent the actual state of Search SDK on each commit.

File: Examples/SearchExamples/Examples.swift [hunk 4]
```diff
@@ -10,6 +10,6 @@ struct Example {
     let screenType: ExampleController.Type
 }
 
-protocol ExampleController: UIViewController {
+public protocol ExampleController: UIViewController {
```
Comment (azarovalex) [inline Examples/SearchExamples/Examples.swift:13]:
```suggestion
protocol ExampleController: UIViewController {
```

File: Examples/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+mapbox_access_token
```
Comment (azarovalex) [inline Examples/.gitignore:1]: I guess the recommended way is to use `~/.mapbox`
Ref: https://docs.mapbox.com/help/troubleshooting/private-access-token-android-and-ios/

File: Examples/SearchExamples/MapsViewController.swift [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
 import MapboxMaps
 import MapboxSearch
 
-class MapsViewController: UIViewController, ExampleController {
+public class MapsViewController: UIViewController, ExampleController {
     let mapView = MapView(frame: .zero)
     lazy var annotationsManager = mapView.annotations.makePointAnnotationManager()
     
-    override func viewDidLoad() {
+    public override func viewDidLoad() {
```
Comment (azarovalex) [inline Examples/SearchExamples/MapsViewController.swift:9]: We don't need to make these public, right?

File: MapboxSearch.podspec [hunk 7]
```diff
@@ -24,5 +24,5 @@ Some iOS platform specifics applies.
```
Comment (azarovalex) [inline MapboxSearch.podspec:24]: pls update `s.ios.deployment_target` to 12.0 in both podspecs

File: README.md [hunk 8]
```diff
@@ -71,8 +71,7 @@ password sk.ey_your_access_token_wit_Read_permission
     1. Create a new file named `mapbox` or `.mapbox` in your home directory with content of your access token. We also support `.mapbox` file in the repository root folder. MapboxSearchDemoApplication will automatically handle this key and insert it in corresponding place.
 
         _Note: Run `pbpaste > ~/.mapbox` in Terminal.app to insert you Pastebord (Command+C buffer) into `.mapbox` in Home directory._
-    1. Open the Workspace, choose `MapboxSearchDemoApplication` project and select "Info" tab for "MapboxSearchDemoApplication" target. Here you can set your accessToken for `MGLMapboxAccessToken` key in "Custom iOS Target Properties" section.
-    1. Provide your accessToken directly in argument named `accessToken` in `SearchDrawer.make(:)` method
+    1. Open the Workspace, choose `MapboxSearchDemoApplication` project and select "Info" tab for "MapboxSearchDemoApplication" target. Here you should set your accessToken for `MGLMapboxAccessToken` key in "Custom iOS Target Properties" section.
```
Comment (azarovalex) [inline README.md:74]: Is `MGLMapboxAccessToken` still used? My understanding that it's a legacy key and `MBXAccessToken` replaced it.

File: Sources/MapboxSearch/PublicAPI/Common/Models/Address/Address.swift [hunk 9]
```diff
@@ -29,14 +29,18 @@ public struct Address: Codable, Hashable {
...
     /// Top-level sub-national administrative features, such as states in the United States or provinces in Canada or China.
     public var region: String?
+
+    public var searchAddressRegion: SearchAddressRegion?
     
     /// Generally recognized countries or, in some cases like Hong Kong, an area of quasi-national administrative status
     /// that has been given a designated country code under ISO 3166-1.
     public var country: String?
 
+    public var searchAddressCountry: SearchAddressCountry?
```
Comment (azarovalex) [inline Sources/MapboxSearch/PublicAPI/Common/Models/Address/Address.swift:42]: let's add some docs to new properties

**Review by azarovalex — APPROVED** [review]

File: README.md (hunk 8 above)
Comment (aokj4ck) [reply README.md:74]: That will be addressed in PR https://github.com/mapbox/mapbox-search-ios/pull/144 / SSDK-523

File: MapboxSearch.podspec (hunk 7 above)
Comment (aokj4ck) [reply MapboxSearch.podspec:24]: Oh that should have been pulled in from SSDK-499. I think I need to adjust my git configuration for the next merge.

File: Examples/.gitignore (hunk 5 above)
Comment (aokj4ck) [reply Examples/.gitignore:1]: I can address that in PR https://github.com/mapbox/mapbox-search-ios/pull/144 / SSDK-523

File: Examples/SearchExamples/Examples.swift (hunk 4 above)
Comment (aokj4ck) [reply Examples/SearchExamples/Examples.swift:13]: This was necessary to build the Examples/SearchExample project with Xcode 14.1 on the iOS 13.6 deployment target.

File: Examples/SearchExamples.xcodeproj/project.pbxproj (hunk 3 above)
Comment (aokj4ck) [reply Examples/SearchExamples.xcodeproj/project.pbxproj:453]: Yeah that's a great idea. I'll split that change into a follow-up ticket / PR to keep this one moving along.

File: Examples/SearchExamples/MapsViewController.swift (hunk 6 above)
Comment (aokj4ck) [reply Examples/SearchExamples/MapsViewController.swift:9]: Yeah we can remove `public` from this class and function 👍

File: Examples/SearchExamples.xcodeproj/project.pbxproj (hunk 3 above)
Comment (aokj4ck) [reply Examples/SearchExamples.xcodeproj/project.pbxproj:453]: → SSDK-525

**Pull request merged**